.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,246 @@
+        uses: actions/upload-artifact@v2
+        with:
+          path: package/dist/*.tar.gz
+          retention-days: 7
+
+  build_sdist_tests:
+    if: "github.repository == 'MDAnalysis/mdanalysis'"
+    name: build test source distribution
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./testsuite
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Build sdist
+        run: pipx run build --sdist
+
+      - name: upload artifacts
+        if: |
+          (github.event_name == 'push' && startsWith(github.ref, 'refs/tags/package')) ||
+          (github.event_name == 'release' && github.event.action == 'published')
+        uses: actions/upload-artifact@v2
+        with:
+          path: testsuite/dist/*.tar.gz
+          retention-days: 7
+
+  upload_testpypi:
+    if: |
+      github.repository == 'MDAnalysis/mdanalysis' &&
+      (github.event_name == 'push' && startsWith(github.ref, 'refs/tags/package'))
+    name: testpypi upload
+    environment: deploy
+    runs-on: ubuntu-latest
+    needs: [build_wheels, build_sdist, build_sdist_tests]
+    steps:
+      - uses: actions/download-artifact@v2
+        with:
+          name: artifact
+          path: dist
+
+      - name: move_test_dist
+        run: |
+          mkdir -p testsuite/dist
+          mv dist/MDAnalysisTests-* testsuite/dist
+
+      - name: upload_source
+        uses: pypa/gh-action-pypi-publish@v1.5.0
+        with:
+          user: __token__
+          password: ${{ secrets.TESTPYPI_API_TOKEN_SRC }}
+          skip_existing: true
+          repository_url: https://test.pypi.org/legacy/
+
+      - name: upload_tests
+        uses: pypa/gh-action-pypi-publish@v1.5.0
+        with:
+          user: __token__
+          password: ${{ secrets.TESTPYPI_API_TOKEN_TEST }}
+          packages_dir: testsuite/dist
+          skip_existing: true
+          repository_url: https://test.pypi.org/legacy/
+ 
+  upload_pypi:
+    if: |
+      github.repository == 'MDAnalysis/mdanalysis' &&
+      github.event_name == 'release' && github.event.action == 'published'
+    name: pypi upload
+    environment: deploy
+    runs-on: ubuntu-latest
+    needs: [build_wheels, build_sdist, build_sdist_tests]
+    steps:
+      - uses: actions/download-artifact@v2
+        with:
+          name: artifact
+          path: dist
+
+      - name: move_test_dist
+        run: |
+          mkdir -p testsuite/dist
+          mv dist/MDAnalysisTests-* testsuite/dist
+
+      - name: upload_source
+        uses: pypa/gh-action-pypi-publish@v1.5.0
+        with:
+          user: __token__
+          password: ${{ secrets.PYPI_API_TOKEN_SRC }}
+
+      - name: upload_tests
+        uses: pypa/gh-action-pypi-publish@v1.5.0
+        with:
+          user: __token__
+          password: ${{ secrets.PYPI_API_TOKEN_TEST }}
+          packages_dir: testsuite/dist
+
+  check_testpypi:
```

Comment:
Really cool that we can automatically test the testpypi distribution. Nice.

Reply:
How long does it take to complete?

Reply:
~ 20 mins per runner

It's quite handy that github actions just lets us deploy to lots of runners if the pool is empty, so an aggregate ~ 6 hours set of tests can be squashed down to 20 mins :)

Reply:
Nice!